src/nodeComponents/click/click.js

```diff
@@ -25,29 +25,36 @@ export default (node) =>
    * .find('Cube_1')
    * .click(true); // you can also pass in `debug` as `true` to see the cursor
    */
-    (debug = false) => {
-    const position = node.getWorldPosition(new Vector3());
+  (debug = false) => {
+    // Find the centre point of the 3D node
+    node.updateMatrixWorld();
+    const box = new THREE.Box3().setFromObject(node);
+    const position = box.getCenter(new THREE.Vector3());
+    position.applyMatrix4(node.matrixWorld);
     const canvas = store.get(C.RENDERER).domElement;
 
-    const screenSpace = worldToScreenSpace(
-      position.x,
-      position.y,
-      position.z,
-      canvas.clientWidth,
-      canvas.clientHeight
-    );
+    const screenSpace = worldToScreenSpace(position.x, position.y, position.z, canvas.clientWidth, canvas.clientHeight);
 
-    simulate(C.CLICK, {
-      clientX: screenSpace.x,
-      clientY: screenSpace.y
-    }, debug);
+    const offset = {
+      left: canvas.getBoundingClientRect().left + window.scrollX,
+      top: canvas.getBoundingClientRect().top + window.scrollY,
+    };
+
+    const eventData = {
+      clientX: screenSpace.x + offset.left,
+      clientY: screenSpace.y + offset.top,
+    };
```

Comment:
I assume we need this, as the current code assumes `canvas` takes up the entire screen space. With these changes, we can ensure we target the canvas size regardless of its position and size within the DOM. is that correct?

If so, this is a great addition. We should generalize this function, similar to `worldToScreenSpace` and include it as a helper. That way, we can use it throughout the application, as the `canvas` is often retrieved.

If you can please update this functionality to be included everywhere the `canvas` sizing and positioning is being queried.

Reply:
@FarhadG: Ok. I did some refactoring. Now, `canvas` is being passed to helper functions. Please check that. I would like to hear your thoughts.

Reply:
@FarhadG: Maybe, `position.x, position.y, position.z` should be replaced with `Vector3`. Function could be more readable and only two parameters could be passed to the function.